type/__packages/manifest

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh -e
+
+os=$(cat "${__global}/explorer/os")
+
+___package() {
+    case "${os}"
+    in
+        (debian|ubuntu)
+            __package_apt "${1}" \
+                --state "${2}" \
+                --purge-if-absent \
+                </dev/null
+            ;;
+        (*)
+            __package "${1}" \
+                --state "${2}" \
+                </dev/null
+            ;;
+    esac
+}
+
+is_present="${__object}/explorer/present"
+should_present="${__object}/parameter/present"
+should_absent="${__object}/parameter/absent"
+
+for packages_list in "${should_present}" "${should_absent}"
+do
+    test ! -f "${packages_list}" && continue
+
+    package_state=$(basename "${packages_list}")
+
+    case "${package_state}"
+    in
+        (present)
+            set -- '-23'
+            ;;
+        (absent)
+            set -- '-12'
+            ;;
+    esac
+
+    grep -Ev '^(#|$)' "${packages_list}" \
+    | sort \
```

Comment:
Please set `LC_ALL=C` for both invocations of `sort(1)`.
People may have servers configured to POSIX locale (or without locales at all) while they have their workstation in their native language which may introduce some nasty to track down bugs.